proto/osmosis/pool-incentives/v1beta1/genesis.proto

```diff
@@ -25,3 +25,28 @@ message GenesisState {
     (gogoproto.moretags) = "yaml:\"pool_to_gauges\""
   ];
 }
+
+// MigrationRecords contains all the links between balancer and concentrated
+// pools.
+//
+// This is copied over from the gamm proto file in order to circumnavigate
+// the circular dependency between the two modules.
+message MigrationRecords {
+  repeated BalancerToConcentratedPoolLink balancer_to_concentrated_pool_links =
+      1 [ (gogoproto.nullable) = false ];
+}
+
+// BalancerToConcentratedPoolLink defines a single link between a single
+// balancer pool and a single concentrated liquidity pool. This link is used to
+// allow a balancer pool to migrate to a single canonical full range
+// concentrated liquidity pool position
+// A balancer pool can be linked to a maximum of one cl pool, and a cl pool can
+// be linked to a maximum of one balancer pool.
+//
+// This is copied over from the gamm proto file in order to circumnavigate
+// the circular dependency between the two modules.
+message BalancerToConcentratedPoolLink {
+  option (gogoproto.equal) = true;
+  uint64 balancer_pool_id = 1;
+  uint64 cl_pool_id = 2;
+}
```

Comment:
This was sadly the cleanest way knew to get passed the circular dependency issue. The only module that pool-incentives had access to that could access gamm's store was superfluid. So I basically called GetAllMigrationInfo in pool-incentives via superfluid and parsed the result into an identical struct. Please lmk if there is a better way to achieve this without a major refactor.

Reply:
this might sound a bit unorthodox but how about we create a separate file that store shared interfaces between modules  

for instance; 
 - gamm
      - shared_keepers.go (includes PoolIncentiveKeeper) 

- pool-incentives 
     - shared_keepers.go (includes GammKeeper)

Reply:
I agree with this! I think it makes it simpler to reason about when we come back around and resolve this with the solution Roman recommended. Added here https://github.com/osmosis-labs/osmosis/pull/5589/commits/e70ac4183287f5fd7626f7dc54fc2f96b6997379